paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheet.kt

```diff
@@ -1516,6 +1535,67 @@ class PaymentSheet internal constructor(
         }
     }
 
+    /**
+     * Options to block certain card brands on the client
+     */
+    sealed class CardBrandAcceptance : Parcelable {
+
+        /**
+         * Card brand categories that can be allowed or disallowed
+         */
+        @Parcelize
+        @ExperimentalCardBrandFilteringApi
+        enum class BrandCategory : Parcelable {
+            /**
+             * Visa branded cards
+             */
+            Visa,
+
+            /**
+             * Mastercard branded cards
+             */
+            Mastercard,
+
+            /**
+             * Amex branded cards
+             */
+            Amex,
+
+            /**
+             * Discover branded cards
+             * **Note**: Encompasses all of Discover Global Network (Discover, Diners, JCB, UnionPay, Elo).
+             */
+            Discover
+        }
+
+        /**
+         * Accept all card brands supported by Stripe
+         */
+        @Parcelize
+        @ExperimentalCardBrandFilteringApi
+        data object All : CardBrandAcceptance()
```

Comment:
I'm wondering if these implementations should be factory methods instead. That way we could leave the implementations internal.

Reply:
Interesting, I pushed up a factory function approach, lmk if that's what you had in mind. Probably don't need to make the `val`'s internal anymore in that case.

Reply:
Seems reasonable to me!